test/command-line.js

```diff
@@ -40,4 +40,49 @@ describe('command line', function () {
       return done();
     });
   });
+  var watchProc;
+  var stdout = '';
+  it('jade --no-debug --client --name-after-file --watch input-file.jade (pass 1)', function (done) {
+    fs.writeFileSync(__dirname + '/temp/input-file.jade', '.foo bar');
+    fs.writeFileSync(__dirname + '/temp/input-file.js', 'throw new Error("output not written (pass 1)");');
+    var args = [__dirname + '/../bin/jade.js', '--no-debug', '--client', '--name-after-file', '--watch', 'input-file.jade']
+    watchProc = cp.spawn('node', args,  {
+      cwd: __dirname + '/temp'
+    });
+
+    watchProc
+      .on('error', done)
+      .stdout.setEncoding('utf8');
+    watchProc.stdout.on('data', function(buf) {
+        stdout += buf;
+        if (/.*rendered.*/.test(stdout)) {
+          stdout = '';
+          var template = Function('', fs.readFileSync(__dirname + '/temp/input-file.js', 'utf8') + ';return inputFileTemplate;')();
+          console.log(template())
+          assert(template() === '<div class="foo">bar</div>');
+
+          watchProc.stdout.removeAllListeners('data');
+          watchProc.removeAllListeners('error');
+          return done();
+        }
+      })
+  });
+  it('jade --no-debug --client --name-after-file --watch input-file.jade (pass 2)', function (done) {
+    fs.writeFileSync(__dirname + '/temp/input-file.js', 'throw new Error("output not written (pass 2)");');
+    fs.writeFileSync(__dirname + '/temp/input-file.jade', '.foo baz');
+
+    watchProc
+      .on('error', done)
+      .stdout.on('data', function(data) {
+        stdout += data;
+        if (/.*rendered.*/.test(stdout)) {
+          watchProc.kill('SIGINT');
```

Comment:
I really like the way this is all done.  The asynchronous nature of file watching is difficult to capture, but I think you nailed it.

My only concern is that if the test times out for some reason, the `watchProc` won't get killed since that only happens in this listener.  I'm thinking we should move it out into an `after` [hook](http://mochajs.org/#hooks) to prevent any zombies.  Something like:
```javascript
describe('jade --watch', function() {  // enclose the whole suite
  var watchProc;
  var stdout = '';
  it('watch test one', function() {})  // first test
  it('watch test two', function() {})  // second test

  after(function() {  // runs after the suite (both tests) finish
    if (watchProc != null) watchProc.kill('SIGINT');
  });
});
```